consensus/core/src/transaction.rs

```diff
@@ -261,19 +261,39 @@ impl TransactionClient {
     /// a receiver to wait on until the transactions has been included in the next block to get proposed. The consumer should
     /// wait on it to consider as inclusion acknowledgement. If the receiver errors then consensus is shutting down and transaction
     /// has not been included to any block.
-    /// If multiple transactions are submitted, the receiver will be signalled when the last transaction is included in the block.
+    /// If multiple transactions are submitted, the method will attempt to bundle them together in a single block. If the total size of
+    /// the transactions exceeds `max_transactions_in_block_bytes`, no transaction will be submitted and an error will be returned instead.
+    /// Similar if transactions exceed `max_transactions_in_block_size` an error will be returned.
     pub(crate) async fn submit_no_wait(
         &self,
         transactions: Vec<Vec<u8>>,
     ) -> Result<oneshot::Receiver<(BlockRef, oneshot::Receiver<BlockStatus>)>, ClientError> {
         let (included_in_block_ack_send, included_in_block_ack_receive) = oneshot::channel();
+
+        let mut bundle_size = 0;
+
+        if transactions.len() as u64 > self.max_transactions_in_block_size {
```

Comment:
In `next()`, `max_consumed_transactions_per_request` is compared against with `>=`, so this should use `>=` as well?

Reply:
Good point. Actually I'll change the `max_consumed_transactions_per_request` to use `>` instead. So maximum will always mean inclusive.